### PR TITLE
Refactor tear fountain logic; remove GenerateRnd loop

### PR DESCRIPTION
**Pull request by vlolteanu**
Addresses https://github.com/diasurgical/devilutionX/issues/2278

### CODE REVIEW

File: Source/objects.cpp [hunk 1]
```diff
@@ -4564,33 +4564,33 @@ bool OperateFountains(int pnum, int i)
...
-			}
-			if (cnt > 1)
+
+		int fromStat = GenerateRnd(4);
+		int toStat = [&]() {
+			int rnd = GenerateRnd(3);
+			if (rnd >= fromStat)
+				rnd++;
+			return rnd;
+		}();
```
Comment (AJenbo) [inline Source/objects.cpp:4574]: What is the lambda giving us here compared to just dowing it directly?
```suggestion
		int toStat = GenerateRnd(3);
		if (toStat >= fromStat)
			toStat++;
```

Comment (vlolteanu) [reply Source/objects.cpp:4574]: Slightly better indication of intent, but otherwise, nothing.

Comment (AJenbo) [reply Source/objects.cpp:4574]: Imo it adds a significant number of cryptic symbols that some new comers may not be familiar with, for something like level objects I think it's important that the code be very approachable for novices.

File: Source/objects.cpp [hunk 2]
```diff
@@ -4564,33 +4564,33 @@ bool OperateFountains(int pnum, int i)
...
+
+		std::pair<int, int> alterations[] = { { fromStat, -1 }, { toStat, 1 } };
+		for (auto i : alterations) {
```
Comment (AJenbo) [inline Source/objects.cpp:4577]:
```suggestion
		for (auto alteration : alterations) {
```

**Review by AJenbo — APPROVED** [review]

File: Source/objects.cpp (hunk 1 above)
Comment (vlolteanu) [reply Source/objects.cpp]: Alright.

I prefer the lambda because toStat >= fromStat isn't an exceptional condition.

**Pull request merged**